### PR TITLE
Fix #513: Result investigateSamples is ignored

### LINKED ISSUE #513: Result investigateSamples is ignored
It seems to me, that when `investigateSample` is called in sync mode, the result is ignored, and `investigation/{id}` is called unnecessarily. Investigate it little bit deeper.

https://github.com/wultra/enrollment-server/blob/979c4614338e5876feff7514f76f2d140f2e864a/enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java#L246

### PR DESCRIPTION
**Pull request by banterCZ**
Make startVerification working at sync mode and store the result immediately

### CODE REVIEW

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java [hunk 1]
```diff
@@ -234,63 +234,6 @@ public List<DocumentVerificationEntity> submitDocuments(DocumentSubmitRequest re
...
-        final DocumentsVerificationResult result = documentVerificationProvider.verifyDocuments(ownerId, uploadIds);
-        final String verificationId = result.getVerificationId();
-        final DocumentVerificationStatus status = result.getStatus();
-        logger.info("Verified documents upload ID: {}, verification ID: {}, status: {}, {}", uploadIds, verificationId, status, ownerId);
-        auditService.auditDocumentVerificationProvider(identityVerification, "Documents verified: {} for user: {}", status, ownerId.getUserId());
-
-        moveToPhaseAndStatus(identityVerification, IdentityVerificationPhase.DOCUMENT_VERIFICATION, IdentityVerificationStatus.IN_PROGRESS, ownerId);
-
-        docVerifications.forEach(docVerification -> {
-            docVerification.setStatus(DocumentStatus.VERIFICATION_IN_PROGRESS);
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:276]: My main objection is this. `VERIFICATION_IN_PROGRESS` is always set, whatever the result is.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 2]
```diff
@@ -131,9 +200,13 @@ private void accept(
...
-        documentVerifications.forEach(docVerification ->
-            auditService.audit(docVerification, "Document accepted at final verification for user: {}", identityVerification.getUserId()));
-        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, ACCEPTED, ownerId);
+
+        final IdentityVerificationPhase phase = identityVerification.getPhase();
+        documentVerifications.forEach(docVerification -> {
+            docVerification.setStatus(DocumentStatus.ACCEPTED);
+            auditService.audit(docVerification, "Document accepted at phase {} for user: {}", phase, identityVerification.getUserId());
+        });
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, phase, ACCEPTED, ownerId);
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:209]: [] Movement to another phase and status should be done only for final verification. For first investigation, we have another idea at #515

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java (hunk 1 above)
Comment (romanstrobl) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:276]: Result of what? We are just starting verification of documents. If you remove this call, then method `checkVerificationResult` will never work.

Comment (romanstrobl) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:276]: Ah, I see, this is the sync vs. async calling, right?

Comment (romanstrobl) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:276]: My concern is what this does to the frontend. The frontend may wait for some time and be still stuck in old state during verification.

Perhaps we can set:

```
moveToPhaseAndStatus(identityVerification, IdentityVerificationPhase.DOCUMENT_VERIFICATION, IdentityVerificationStatus.IN_PROGRESS, ownerId);
```
and 

```          docVerification.setStatus(DocumentStatus.VERIFICATION_IN_PROGRESS);
```

in another transaction before starting the verification? This way the frontend will obtain correct state during verification. After your change there will be no way to find out for frontend that verification is in progress, right?

Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:276]: > It's quite a big change, so we'll need to test this thoroughly.

Exactly. I do not push this change. I would like to postpone it, if possible, when we stabilize the current issues.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 3]
```diff
@@ -105,14 +116,80 @@ public void executeFinalDocumentVerification(final IdentityVerificationEntity id
...
+    /**
+     * Starts the verification process
+     *
+     * @param ownerId Owner identification.
+     * @param identityVerification Identity verification.
+     * @throws RemoteCommunicationException In case of remote communication error.
+     * @throws DocumentVerificationException In case of business logic error.
+     */
+    @Transactional
+    public void startVerification(OwnerId ownerId, IdentityVerificationEntity identityVerification) throws DocumentVerificationException, RemoteCommunicationException, OnboardingProcessException {
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:132]: Mainly moved from https://github.com/wultra/enrollment-server/pull/514/files#diff-0454fa519907321b8ba689699b8fc5e5d8835ea5172e7d29137e24ef4e9eb0baL239

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 4]
```diff
@@ -105,14 +116,80 @@ public void executeFinalDocumentVerification(final IdentityVerificationEntity id
...
+            logger.debug("Selfie photos verification disabled, changing selfie document status to ACCEPTED, {}", ownerId);
+            selfiePhotoVerifications.forEach(selfiePhotoVerification -> {
+                selfiePhotoVerification.setStatus(DocumentStatus.ACCEPTED);
+                selfiePhotoVerification.setTimestampLastUpdated(ownerId.getTimestamp());
+                auditService.audit(selfiePhotoVerification, "Selfie document accepted for user: {}", ownerId.getUserId());
+            });
+            documentVerificationRepository.saveAll(selfiePhotoVerifications);
+        }
+
+        changeDocumentVerificationStatusAndVerificationId(docVerifications, result, identityVerification, ownerId);
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:167]: The main reason of change is here. We change the status of the documents based on the response from the provider. It used to be (whatever the result was)
```java
docVerifications.forEach(docVerification -> {
            docVerification.setStatus(DocumentStatus.VERIFICATION_IN_PROGRESS);
```

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java (hunk 1 above)
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:269]: `moveToPhaseAndStatus(identityVerification, IdentityVerificationPhase.DOCUMENT_VERIFICATION, IdentityVerificationStatus.IN_PROGRESS, ownerId);` was kept.

Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java:269]: `VerificationProcessingBatchService` kept to enable async in the future.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java (hunk 4 above)
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:167]: 🤔 May be this is too complicated and calling `verificationProcessingService.processVerificationResult` could be enough.

**Comment by banterCZ** [comment]
Simplified at https://github.com/wultra/enrollment-server/pull/536

**Pull request closed without merging**